crypto-pgp/src/main/kotlin/dev/msfjarvis/aps/data/crypto/GPGKeyManager.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2014-2021 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package dev.msfjarvis.aps.data.crypto
+
+import androidx.annotation.VisibleForTesting
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.runCatching
+import com.proton.Gopenpgp.crypto.Crypto
+import java.io.File
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+
+public class GPGKeyManager(filesDir: String, private val dispatcher: CoroutineDispatcher) :
+  KeyManager<GPGKeyPair> {
+
+  private val keyDir = File(filesDir, KEY_DIR_NAME)
+  override suspend fun addKey(key: GPGKeyPair): Result<String, Throwable> =
+    withContext(dispatcher) {
+      runCatching {
+        if (!keyDirExists()) error("Key directory does not exist")
+        val keyFile = File(keyDir, "${key.getKeyId()}.$KEY_EXTENSION")
+        if (keyFile.exists()) {
+          if (!keyFile.delete()) error("Couldn't delete existing key file with the same name")
+        }
+
+        keyFile.writeBytes(key.getPrivateKey())
+
+        key.getKeyId()
+      }
+    }
+
+  override suspend fun removeKey(key: GPGKeyPair): Result<String, Throwable> =
+    withContext(dispatcher) {
+      runCatching {
+        if (!keyDirExists()) error("Key directory does not exist")
+        val keyFile = File(keyDir, "${key.getKeyId()}.$KEY_EXTENSION")
+        if (keyFile.exists()) {
+          if (!keyFile.delete()) error("Couldn't delete key file")
+        }
+
+        key.getKeyId()
```

Comment:
Does it make sense to be returning this? We only really care if this throws an error, so the happy path should just be `Unit`, no?

Reply:
I just want to keep the API consistent b/w `addKey` and `removeKey`. We can make both of them return `Unit` or `KeyPair`. Which one do you prefer?

Reply:
It's fine, let it be KeyPair.